src/heroic.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::HashMap;
+
+use crate::{
+    config::RootsConfig,
+    manifest::{Manifest, Store},
+    prelude::{normalize_title, StrictPath},
+};
+
+// Deserialization of Heroic gog_store/installed.json
+#[derive(serde::Deserialize)]
+struct HeroicInstalledGame {
+    #[serde(rename = "appName")]
+    game_id: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct HeroicInstalled {
+    installed: Vec<HeroicInstalledGame>,
+}
+
+// Deserialization of Heroic gog_store/library.json
+#[derive(serde::Deserialize)]
+struct GogLibraryGame {
+    app_name: String,
+    title: String,
+}
+#[derive(serde::Deserialize)]
+struct GogLibrary {
+    games: Vec<GogLibraryGame>,
+}
+
+// Deserialization of Legendary legendary/installed.json
+#[derive(serde::Deserialize)]
+struct LegendaryInstalledGame {
+    #[serde(rename = "app_name")]
+    game_id: String,
+    title: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct LegendaryInstalled(HashMap<String, LegendaryInstalledGame>);
+
+// TODO.2022-10-10 heroic: windows location for legendary
+// TODO.2022-10-15 heroic: make relative to heroic root in question!
+const LEGENDARY_PATHS: &[&str] = &[
+    "~/.config/legendary",
+    // TODO.2022-10-20 heroic: flatpak install is not supported yet
+    // "~/.var/app/com.heroicgameslauncher.hgl/config/legendary",
+];
+
+#[derive(Clone, Default, Debug)]
+pub struct HeroicGames {
+    games: HashMap<String, StrictPath>,
+    normalized_to_official: HashMap<String, String>,
+}
+
+impl HeroicGames {
+    pub fn get(&self, game: &String) -> Option<&StrictPath> {
+        self.games.get(game)
+    }
+
+    pub fn scan(roots: &[RootsConfig], manifest: &Manifest) -> Self {
+        let mut instance = HeroicGames {
+            normalized_to_official: manifest
+                .0
+                .keys()
+                .map(|title| (normalize_title(title), title.clone()))
+                .collect(),
+            ..Default::default()
+        };
+
+        roots.iter().for_each(|root: &RootsConfig| {
```

Comment:
I prefer regular for-loops over `for_each`. If you just needed to avoid an ownership issue, you can borrow:

```
for root in &roots {}
```

Reply:
Yes, much better and concise style!  I've read the rust book chapter on loops after I've written this line :)